File: 2418-sort-the-people/2418-sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        // Create a vector of pairs (height, name)
        vector<pair<int, string>> people;
        
        // Populate the people vector
        for (int i = 0; i < names.size(); i++) {
            people.emplace_back(heights[i], names[i]);
        }
        
        // Sort the people vector in descending order based on height
        sort(people.begin(), people.end(), [](const pair<int, string>& a, const pair<int, string>& b) {
            return a.first > b.first;
        });
        
        // Extract the sorted names
        vector<string> sortedNames;
        for (const auto& person : people) {
            sortedNames.push_back(person.second);
        }
        
        return sortedNames;
    }
};

File: 2418-sort-the-people/2418-sort-the-people.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
var sortPeople = function(names, heights) {
    // Create a list of tuples (height, name)
    let people = names.map((name, index) => [heights[index], name]);
    
    // Sort the list in descending order based on height
    people.sort((a, b) => b[0] - a[0]);
    
    // Extract the names from the sorted list
    let sortedNames = people.map(person => person[1]);
    
    return sortedNames;
};
